### ISSUE TITLE
[citeproc] Regression with respect to duplicate references

### ISSUE THREAD

**Issue by mpark** [jgm/citeproc#36]
Following up on jgm/pandoc#6951 and 7d799bfcda749e4a3ad6fcae59ac5ccb80b77ffd

Pandoc 2.11

`foo.md`:
```
[@foo]

---
references:
  - id: foo
    title: inline
...
```

`index.yaml`:
```
references:
  - id: foo
    title: external
```

With Pandoc 2.9:
```
pandoc foo.md --bibliography index.yaml
<p><span class="citation" data-cites="foo">(“Inline,” n.d.)</span></p>
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-foo">
<p>“Inline.” n.d.</p>
</div>
</div>
```

At 7d799bfcda749e4a3ad6fcae59ac5ccb80b77ffd, it seems to use "external", although it also produces multiple entries:
```
pandoc foo.md --citeproc --bibliography index.yaml
<p><span class="citation" data-cites="foo">(<span>“External,”</span> n.d.)</span></p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-foo" class="csl-entry" role="doc-biblioentry">
<p><span>“External.”</span> n.d.</p>
</div>
<div id="ref-foo" class="csl-entry" role="doc-biblioentry">
<p>———. n.d.</p>
</div>
</div>
```

Similarly, with multiple `--bibliography` arguments.
`index2.yaml`:
```
references:
  - id: foo
    title: external2
```

With Pandoc 2.9:
```
pandoc foo.md --bibliography index.yaml --bibliography index2.yaml
<p><span class="citation" data-cites="foo">(“External,” n.d.)</span></p>
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-foo">
<p>“External.” n.d.</p>
</div>
</div>
```

At 7d799bfcda749e4a3ad6fcae59ac5ccb80b77ffd:
```
pandoc foo.md --citeproc --bibliography index.yaml --bibliography index2.yaml
<p><span class="citation" data-cites="foo">(<span>“External2,”</span> n.d.)</span></p>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-foo" class="csl-entry" role="doc-biblioentry">
<p><span>“External2.”</span> n.d.</p>
</div>
<div id="ref-foo" class="csl-entry" role="doc-biblioentry">
<p>———. n.d.</p>
</div>
</div>
```

**Comment by jgm** [comment]
```
% cat test.json
{ "citations":
[ [ {"id":"foo"} ] ],
  "references":
[
  {
    "id": "foo",
    "type": "book",
    "title": "Foo1"
  },
  {
    "id": "foo",
    "type": "book",
    "title": "Foo2"
  }
]

}
```
```
% % citeproc test.json --style ../styles/chicago-author-date.csl 
{
  "citations": [
    "(<i>Foo2</i>, n.d.)"
  ],
  "bibliography": [
    [
      "foo",
      "<i>Foo2</i>. n.d."
    ],
    [
      "foo",
      "———. n.d."
    ]
  ],
  "warnings": []
}
```

**Comment by jgm** [comment]
So there are four separate issues here:

1.  When there are multiple references with the same id, which one should take precedence?  Currently it's the one that comes later in the list.  Is that sensible or should it be changed?  This has to do with construction of the reference map from the list of references.
2. Should citeproc issue a warning when multiple references are given for the same id?
3. When we have multiple references for the same id, what should we do about the bibliography?  Presumably only the one that gets into the reference map (see 1) should be included in the bibliography.
4. [Pandoc issue]:  regardless of how 1 is resolved, we might want to keep the pandoc-citeproc behavior of preferring inline to external references when there is an id conflict.

**Comment by mpark** [comment]
1. I think that the ones that come later superseding the previous entries is sensible. Similar to how later `--defaults` flags will override earlier `--defaults` (for non-extending fields).
2. Maybe? Although if it did issue warnings, I'd specify `--quiet` since it'd likely be spammy for my use case.
3. Yes, I think that only the one that gets into the reference map should get a bibliography.
4. Yes, I think inline references should override external references.

**Comment by jgm** [comment]
Closed by 5fb55778c1d969e662f96efa0e123241a161f60d
